2.7.0/final.md

```diff
@@ -0,0 +1,958 @@
+We now track a mapping between the nodes in the pre-grad and post-grad graph. See the issue for an example frontend to visualize the transformations. To update your `GraphTransformObserver` subclasses, instead of overriding `on_node_creation` and `on_node_erase`, there are new functions `get_node_creation_hook`, `get_node_erase_hook`, `get_node_replace_hook` and `get_deepcopy_hook`. These are registered on the `GraphModule` member of the `GraphTransformObserver` upon entry and exit of a `with` block
+
+Version 2.6.0
+
+```python
+class MyPrintObserver(GraphTransformObserver):
+    def on_node_creation(self, node: torch.fx.Node):
+        print(node)
+```
+Version 2.7.0
+```python
+class MyPrintObserver(GraphTransformObserver):
+    def get_node_creation_hook(self):
+        def hook(node: torch.fx.Node):
+            print(node)
+        return hook
+```
+
+### `torch.onnx.dynamo_export` now uses the ExportedProgram logic path (#137296)
+
+Users using the `torch.onnx.dynamo_export` API may see some `ExportOptions` become
+unsupported due to an internal switch to use `torch.onnx.export(..., dynamo=True)`: `diagnostic_options`, `fake_context` and `onnx_registry` are removed/ignored by `ExportOptions`. Only `dynamic_shapes` is retained.
+
+Users should move to use the `dynamo=True` option on `torch.onnx.export` as
+`torch.onnx.dynamo_export` is now deprecated. Leverage the [`dynamic_shapes`](https://pytorch.org/docs/stable/export.html#torch.export.export) argument in `torch.onnx.export` for specifying dynamic shapes on the model.
+
+Version 2.6.0
+
+```py
+torch.onnx.dynamo_export(model, *args, **kwargs)
+```
+
+Version 2.7.0
+
+```py
+torch.onnx.export(model, args, kwargs=kwargs, dynamo=True)
+```
+
+### `LRScheduler.print_lr()` along with the `verbose` kwarg to the LRScheduler constructor has been deprecated since release 2.2. (#147301)
+
+Please use `LRScheduler.get_last_lr()` to access the learning rate instead.`print_lr` and `verbose` were confusing, not properly documented and were little used, as described in #99270, so we deprecated them in 2.2. Now, we complete the deprecation by removing them completely. To access and print the learning rate of a LRScheduler:
+
+Version 2.6.0
+```
+optim = ...
+lrsched = torch.optim.lr_scheduler.ReduceLROnPlateau(optim, verbose=True)
+// lrsched will internally call print_lr
+```
+
+Version 2.7.0
+```
+optim = ...
+lrsched = torch.optim.lr_scheduler.ReduceLROnPlateau(optim)
+print(lrsched.get_last_lr())
+```
+
+### Change `torch.Tensor.new_tensor()` be on the given Tensor's device by default ([#144958](https://github.com/pytorch/pytorch/pull/144958))
+
+This function was always creating the new Tensor on the "cpu" device and will now use the same device as the current Tensor object. This behavior is now consistent with other `.new_*` methods.
+
+### Please use `torch.export.export_for_training` instead of `capture_pre_autograd_graph` to export the model for pytorch 2 export quantization (#139505)
```

Comment:
This is a deprecation?

Reply:
cc @jerryzh168

Reply:
What is the full path to capture_pre_autograd_graph?

Reply:
`torch._export.capture_pre_autograd_graph`

Reply:
If you've broken it, how is this a deprecation?

Reply:
moving to BC breaking